src/ViewModel/GraphqlViewModel.php

```diff
@@ -0,0 +1,44 @@
+<?php declare(strict_types=1);
+/**
+ * Hyvä Themes - https://hyva.io
+ * Copyright © Hyvä Themes. All rights reserved.
+ * See LICENSE.md for license details
+ */
+
+namespace Hyva\GraphqlViewModel\ViewModel;
+
+use Magento\Framework\Event\ManagerInterface as EventManager;
+use Magento\Framework\View\Element\Block\ArgumentInterface;
+
+class GraphqlViewModel implements ArgumentInterface
+{
+    /**
+     * @var EventManager
+     */
+    private $eventManager;
+
+    public function __construct(EventManager $eventManager)
+    {
+        $this->eventManager = $eventManager;
+    }
+
+    public function query(string $name, string $query): string
+    {
+        return $this->toString($this->toAst($name, $query));
+    }
+
+    public function toAst(string $name, string $query): \GraphQL\Language\AST\DocumentNode
```

Comment:
This method is public so it is pluginazable.
Maybe it would be better to have that on a different class, not the view model, because it is not intended to be called from templates or custom code.

Reply:
Sounds reasonable, but is there really a need for plugins since we have the event? What's the use case?

Reply:
Some people prefer plugins is the only use case I can think of...

Reply:
I think we can be opinionated here, but it's not up to me to decide :)